components/Searchbar.js

```diff
@@ -0,0 +1,169 @@
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useRef, useState, useEffect } from 'react';
+
+export default function SearchBar({ books }) {
+	// Hooks for search bar
+	const searchRef = useRef(null);
+
+	// router
+	const router = useRouter();
+
+	// For click to close input. Not in use right now
+	/**
+	 * @param {HTMLElement} child 
+	 * @param {HTMLElement} parent 
+	 */
+	/*
+	const isDeepChildOf = (child, parent) => {
+		console.log(child.parentNode, parent);
+		if (child == parent) {
+			return true;
+		} else if (child === document.body || child === document.documentElement || document.head || window) {
+			return false;
+		} else {
+			return isDeepChildOf(child.parentNode, parent);
+		}
+	}
+	*/
+
+	// event listeners on page load
+	useEffect(() => {
+		window.addEventListener('keydown', (e) => {
+			if (e.key === '/' || (e.key === 'k' && e.ctrlKey)) {
+				if (searchRef.current && e.target !== searchRef.current) {
+					e.preventDefault();
+					searchRef.current.focus();
+				}
+			} else if (e.key === 'Escape') {
+				e.preventDefault();
+				searchRef.current.value = '';
+				searchRef.current.blur();
+				setSearchItems('hidden');
+				setSelected(null);
+				setInputValue('');
+			}
+		});
+	}, []);
+
+	// list of books to be shown under search
+	const [searchItems, setSearchItems] = useState('hidden');
+	// selected book in list
+	const [selected, setSelected] = useState(null);
+	// current value of input
+	const [inputValue, setInputValue] = useState('');
+
+	return (
+		<div className="relative my-auto h-6 w-48">
+			<div
+				className="absolute ml-3 bg-white overflow-y-auto rounded-xl z-20 shadow-2xl"
+				onKeyDown={
```

Comment:
Something like "Allow users to arrow up and down to select search results and hit enter to select a result" would be really helpful here.